packages/framework/src/client/__tests__/index.spec.js

```diff
@@ -37,10 +37,39 @@ describe( 'ApiClient', () => {
 	};
 
 	it( 'should initialize to empty state', () => {
-		const apiClient = new ApiClient( emptyApiSpec );
+		const apiClient = new ApiClient( emptyApiSpec, null );
 		expect( apiClient.state ).toEqual( {} );
 	} );
 
+	it( 'should stop scheduler when hidden and update scheduler when visible', () => {
+		let eventListener = null;
+		const doc = {
+			addEventListener: ( name, listener ) => {
+				eventListener = listener;
+			},
+			visibilityState: 'visible',
+		};
+		const apiClient = new ApiClient( emptyApiSpec, doc );
+		apiClient.scheduler = {
+			stop: jest.fn(),
+			updateDelay: jest.fn(),
+		};
+
+		expect( eventListener ).not.toBe( null );
+
+		// Simulate tab hidden
+		doc.visibilityState = 'hidden';
+		eventListener()
+
+		expect( apiClient.scheduler.stop ).toHaveBeenCalledTimes( 1 );
+
+		// Simulate tab shown again
+		doc.visibilityState = 'visible';
+		eventListener()
+
+		expect( apiClient.scheduler.updateDelay ).toHaveBeenCalledTimes( 1 );
```

Comment:
I wonder if it would make sense here to make sure that `apiClient.scheduler.stop` has still only be called once? I don't think it's possible that `stop()` has been called again _if the event handler has only been called twice_, but what if sometime in the future the event handler will be called a third (or more) time for whatever reason?

I'll leave defer to you on whether to add this check or not 🙂 .

It might also just make sense to maybe mock the event listener in some way and ensure that the `visibilitychange` event listener has been called twice, and only twice?

Reply:
I originally had a test that mocked the visibility change listener, but I felt it was redundant with this test.
I did, however, add a check for stop to be only called once. Thanks!